velox/functions/prestosql/Comparisons.h

```diff
@@ -17,24 +17,74 @@
 
 #include "velox/common/base/CompareFlags.h"
 #include "velox/functions/Macros.h"
+#include "velox/functions/prestosql/types/TimestampWithTimeZoneType.h"
 
 namespace facebook::velox::functions {
-#define VELOX_GEN_BINARY_EXPR(Name, Expr, TResult)                \
-  template <typename T>                                           \
-  struct Name {                                                   \
-    VELOX_DEFINE_FUNCTION_TYPES(T);                               \
-    template <typename TInput>                                    \
-    FOLLY_ALWAYS_INLINE void                                      \
-    call(TResult& result, const TInput& lhs, const TInput& rhs) { \
-      result = (Expr);                                            \
-    }                                                             \
+
+namespace {
+template <typename T>
+struct TimestampWithTimezoneComparisonSupport {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Convert TimestampWithTimezone from original timezone to GMT in milliseconds
+  FOLLY_ALWAYS_INLINE
+  int64_t toGMTMillis(
+      const arg_type<TimestampWithTimezone>& timestampWithTimezone) {
+    const int64_t milliseconds = *timestampWithTimezone.template at<0>();
```

Comment:
I know there were some discussions about having nano second precision  , I am wondering if the change to timezone and subsequent conversion to millis will cause precision loss.

Reply:
Thanks for calling out this! here milliseconds was used mainly because the corresponding function in presto is converting it to milliseconds by bit shifting https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/type/DateTimeEncoding.java#L50-L54. 
My thoughts here is that we can go with nanoseconds and have a better precision but would it cause inconsistency in result